common/src/main/java/com/revenuecat/purchases/common/offlineentitlements/PurchasedProductsFetcher.kt

```diff
@@ -57,11 +57,13 @@ class PurchasedProductsFetcher(
         val purchaseAssociatedToProduct = allPurchases.first { it.skus[0] == productIdentifier }
         val isActive = activeProducts.contains(productIdentifier)
         val expirationDate = getExpirationDate(isActive, purchaseAssociatedToProduct)
+        val mapping = productEntitlementMapping?.mappings?.get(productIdentifier)
```

Comment:
Currently `productIdentifier` will never include the `basePlanId`, so currently we will always grant the entitlements that the backwards compatible plan would give us... We can probably improve this though. @vegaro has a TODO for this in [#959 ](https://github.com/RevenueCat/purchases-android/pull/959#discussion_r1170165644)